neo/rawio/medrawio.py

```diff
@@ -0,0 +1,357 @@
+        stream_segment_contigua = self._stream_info[stream_index]['contigua']
+        return (stream_segment_contigua[seg_index]['end_index'] - stream_segment_contigua[seg_index]['start_index']) + 1
+
+    def _get_signal_t_start(self, block_index, seg_index, stream_index):
+        return (self._seg_t_starts[seg_index] + self._session_time_offset) / 1e6
+
+    def _get_analogsignal_chunk(self, block_index, seg_index, i_start, i_stop,
+                                stream_index, channel_indexes):
+        
+        import dhn_med_py
+        from dhn_med_py import MedSession
+
+        # Correct for None start/stop inputs
+        if i_start is None:
+            i_start = 0
+        if i_stop is None:
+            i_stop = self.get_signal_size(block_index=block_index, seg_index=seg_index,
+                                          stream_index=stream_index)
+            
+        # Check for invalid start/stop inputs
+        if i_start < 0 or i_stop > self.get_signal_size(block_index=block_index, seg_index=seg_index, stream_index=stream_index):
+            raise IndexError("MED read error: Too many samples requested!")
+        if i_start > i_stop:
+            raise IndexError("MED read error: i_start (" + i_start + ") is greater than i_stop (" + i_stop + ")")
+
+        num_channels = 0
+        if channel_indexes is None:
+            self.sess.set_channel_inactive('all')
+            self.sess.set_channel_active(self._stream_info[stream_index]['raw_chans'])
+            num_channels = len(self._stream_info[stream_index]['raw_chans'])
+            self.sess.set_reference_channel(self._stream_info[stream_index]['raw_chans'][0])
+        else:
+            if any(channel_indexes < 0):
+                raise IndexError(f'Can not index negative channels: {channel_indexes}')
+            # Set all channels to be inactive, then selectively set some of them to be active
+            self.sess.set_channel_inactive('all')
+            for i, channel_idx in enumerate(channel_indexes):
+                num_channels += 1
+                self.sess.set_channel_active(self._stream_info[stream_index]['raw_chans'][channel_idx])
+            self.sess.set_reference_channel(self._stream_info[stream_index]['raw_chans'][channel_indexes[0]])
+              
+        # Return empty dataset if start/stop samples are equal
+        if i_start == i_stop:
+            raw_signals = np.zeros((0, num_channels), dtype='int32')
+            return raw_signals
+
+        # Otherwise, return the matrix 2D array returned by the MED library
+        start_sample_offset = self._stream_info[stream_index]['contigua'][seg_index]['start_index']
+        self.sess.read_by_index(i_start + start_sample_offset, i_stop + start_sample_offset)
+        
+        # Create "sample_major" 2D numpy array from the result of read_by_index()
+        samps_returned = len(self.sess.data['channels'][0]['data'])
+        raw_signals = np.array([], dtype=np.int32)
```

Comment:
I guess theses two lines can b eremoved no ?

Reply:
Yes, you are correct, they are no longer necessary.  I removed the lines.